### PR TITLE
GHA workflow

### PR DESCRIPTION
**Pull request by skripted-io**
I used @cwillisf GHA template for scratch-vm as a basis. 

Additions/changes:
- A release version ID is generated in the form of "0.2.0-prerelease.20230929153610"
- Instead of "npx --no -- semantic-release" we are using "npm publish --tag $NPM_TAG"
- The current commit is tagged with the release version after successful publish

~~That last bit wasn't tested, because npm publish has a --dry-run flag which can be removed once this code is approved. If the tagging then fails I need to look at that again.~~ fixed with a condition.

### CODE REVIEW

File: .github/workflows/update-i18n.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Update i18n
```
Comment (cwillisf) [inline .github/workflows/update-i18n.yml]: This repo doesn't need this workflow since it doesn't have any localized UI

File: .gitignore [hunk 2]
```diff
@@ -12,3 +12,6 @@ dist/*
 # Editors
 /#*
 *~
+
+# Act
+.secrets
```
Comment (cwillisf) [inline .gitignore:17]: Good idea! I'll add this in some of my work...

File: .github/workflows/ci-cd.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+              run: npm run build
+
+            - name: Generate release version
+              run: |
+                  export RELEASE_TIMESTAMP=$(date +'%Y%m%d%H%M%S')
+                  export VPKG=$($(npm bin)/json -f package.json version)
+                  export VERSION=${VPKG}-prerelease.${RELEASE_TIMESTAMP}
+
+                  echo "RELEASE_VERSION=${VERSION}" >> $GITHUB_ENV
+                  echo "NPM_TAG=latest" >> $GITHUB_ENV
```
Comment (cwillisf) [inline .github/workflows/ci-cd.yml]: Looks like this is missing the hotfix handling from the CircleCI config: https://github.com/scratchfoundation/scratch-svg-renderer/blob/0705358502413eba9ab9859eed18edd4a420b23e/.circleci/config.yml#L42-L59

Sorry this is so inconsistent between repositories, but this repo in particular has been a source of hotfixes in the past so I think it's important to retain that support

File: package-lock.json
Comment (cwillisf) [inline package-lock.json:1]: Thanks for moving to `npm ci` and adding a package lockfile! 😎

**Review by cwillisf — CHANGES_REQUESTED** [review]
Great progress! I see a few important differences compared to the CCI config.

File: .github/workflows/update-i18n.yml (hunk 1 above)
Comment (skripted-io) [reply .github/workflows/update-i18n.yml]: Got it. Thnx!

**Comment by cwillisf** [comment]
Oh, right... looks like the commitlint step is complaining about the commit messages not following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary). Most of the front-end & editor repositories use that format now.

File: .github/workflows/ci-cd.yml (hunk 3 above)
Comment (skripted-io) [reply .github/workflows/ci-cd.yml]: Thanks, yeah I think I ignored it for convenience :-D 
on it!

File: .github/workflows/ci-cd.yml [hunk 4]
```diff
@@ -0,0 +1,79 @@
+name: CI/CD
+
+on:
+    workflow_dispatch: # Allows you to run this workflow manually from the Actions tab
+    pull_request: # Runs whenever a pull request is created or updated
+    push: # Runs whenever a commit is pushed to the repository
+        branches: [main, develop, hotfix/*]
```
Comment (cwillisf) [inline .github/workflows/ci-cd.yml]: Ah, that makes sense. Thanks!

**Review by cwillisf — COMMENTED** [review]
Sorry, I had this review pending and forgot to submit it.
This looks like it'll work! Thanks :)

**Review by cwillisf — APPROVED** [review]

File: .github/workflows/ci-cd.yml (hunk 4 above)
Comment (apple502j) [reply .github/workflows/ci-cd.yml]: The branch is actually called "master" here.

**Pull request merged**